mobile/src/reducers/people/people.reducer.ts

```diff
@@ -0,0 +1,147 @@
+    headDepartment: Department;
+    departments: Department[];
+    departmentsHeads: Employee[];
+    employees: Employee[];
+    departmentsHeadsIds: string[];
+    departmentsBranch: DepartmentsTreeNode[];
+    departmentsTree: DepartmentsTree;
+}
+
+const initState: PeopleState = {
+    headDepartment: null,
+    departments: [],
+    departmentsHeads: [],
+    employees: [],
+    departmentsHeadsIds: [],
+    departmentsBranch: [],
+    departmentsTree: null
+};
+
+function departmentsTreeFor(departments: Department[], employees: Employee[]) {
+    const topLevelDepartment: Department = departments.filter((department) => department.isHeadDepartment === true)[0];
+    const topLevelEmployee: Employee = employees.filter((employee) => employee.employeeId === topLevelDepartment.chiefId)[0];
+    const departmentsTree: DepartmentsTree = {
+        root: {
+            departmentId: topLevelDepartment.departmentId,
+            departmentAbbreviation: topLevelDepartment.abbreviation,
+            departmentChiefId: topLevelDepartment.chiefId,
+            head: topLevelEmployee,
+            parent: null,
+            children: childrenNodes(topLevelDepartment, departments, employees),
+            subordinates: employees.filter((employee) => employee.departmentId === topLevelDepartment.departmentId)
+        }
+    };
+
+    return departmentsTree;
+}
+
+function onlyUnique(value: string, index: number, self: string[]) { 
+    return self.indexOf(value) === index;
+}
+
+function childrenNodes(headDeaprtment: Department, departments: Department[], employees: Employee[]) {
+    var nodes: DepartmentsTreeNode[] = [];
+    const siblings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === headDeaprtment.departmentId);
+    siblings.forEach(department => {
+        const employee: Employee = employees.filter((emp) => emp.employeeId === department.chiefId)[0];
+        const subSiblings = departments.filter((subDepartment) => subDepartment.parentDepartmentId === department.departmentId);
+        nodes.push({
+            departmentId: department.departmentId,
+            departmentAbbreviation: department.abbreviation,
+            departmentChiefId: department.chiefId,
+            head: employee,
+            parent: headDeaprtment,
+            children: subSiblings.length > 0 ? childrenNodes(department, departments, employees) : null,
+            subordinates: employees.filter((emp) => emp.departmentId === department.departmentId)
+        });
+    });
+
+    return nodes;
+}
+
+export const peopleReducer: Reducer<PeopleState> = (state = initState, action: PeopleActions | LoadEmployeeFinished | LoadDepartmentsFinished) => {
+    switch (action.type) {
+        case 'LOAD_EMPLOYEE_FINISHED':
+            var heads = state.departmentsHeads;
+
+            if (state.departmentsHeadsIds.indexOf(action.employee.employeeId) > -1 && heads.filter((employee) => employee.employeeId === action.employee.employeeId).length === 0) {
```

Comment:
`find` instead of `filter`, filter iterates the whole array, even the item has been found. Check all places